test/org/zaproxy/zap/authentication/UsernamePasswordAuthenticationCredentialsTest.java

```diff
@@ -0,0 +1,115 @@
+package org.zaproxy.zap.authentication;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.zaproxy.zap.extension.api.ApiResponse;
+import org.zaproxy.zap.extension.api.ApiResponseSet;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+
+
+public class UsernamePasswordAuthenticationCredentialsTest {
+
+    private UsernamePasswordAuthenticationCredentials usernamePasswordAuthenticationCredentials;
+    private ApiResponseSet apiResponseSet;
+    private String username = "username";
+    private String password = "password";
+
+    @Before
+    public void setUp() {
+        this.usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+    }
+
+    @Test
+    public void shouldNotUsernameBeEmptyOrNull() throws Exception {
+        // Given
+        String username = usernamePasswordAuthenticationCredentials.getUsername();
+
+        // When/Then
+        assertThat(username, not(isEmptyOrNullString()));
+    }
+
+    @Test
+    public void shouldNotPasswordBeEmptyOrNull() throws Exception {
+        // Given
+        String password = usernamePasswordAuthenticationCredentials.getPassword();
+
+        // When/Then
+        assertThat(password, not(isEmptyOrNullString()));
+    }
+
+    @Test
+    public void shouldReturnTureIfIsConfigured() throws Exception {
```

Comment:
There's a typo in "True", but how about "shouldBeConfiguredIfItHasNonNullUsernameAndPassword"? That tells when it is expected to be configured.